### PR TITLE
util translate batch 3

### PR DESCRIPTION
**Pull request by d2r**

### CODE REVIEW

File: storm-core/src/clj/org/apache/storm/stats.clj [hunk 1]
```diff
@@ -1376,6 +1377,12 @@
   {:emitted (aggregate-counts (map #(.get_emitted ^ExecutorStats %) stats-seq))
    :transferred (aggregate-counts (map #(.get_transferred ^ExecutorStats %) stats-seq))})
 
+(defn- collectify
+  [obj]
+  (if (or (sequential? obj) (instance? Collection obj))
+    obj
+    [obj]))
```
Comment (d2r) [inline storm-core/src/clj/org/apache/storm/stats.clj]: Add comment or move back to utils. Java's strong type checking will make this unnecessary.

File: storm-core/src/clj/org/apache/storm/testing.clj [hunk 2]
```diff
@@ -98,6 +99,29 @@
...
+        vars (map first settings)
+        savevals (vec (mapcat (fn [t v] [t v]) tmpvars vars))
+        setters (for [[v s] settings] `(set-var-root ~v ~s))
+        restorers (map (fn [v s] `(set-var-root ~v ~s)) vars tmpvars)]
+    `(let ~savevals
+       ~@setters
+       (try
+         ~@body
+         (finally
+           ~@restorers)))))
```
Comment (d2r) [inline storm-core/src/clj/org/apache/storm/testing.clj]: Move these back to utils

File: storm-core/src/clj/org/apache/storm/testing.clj [hunk 3]
```diff
@@ -700,3 +740,53 @@
      (try
        (.get f# ~millis ~unit)
        (finally (future-cancel f#)))))
+
+(defmacro mock-java-static
```
Comment (d2r) [inline storm-core/src/clj/org/apache/storm/testing.clj]: This is needed in Java too, so it should be translated as an AutoCloseable.

File: storm-core/src/clj/org/apache/storm/testing.clj (hunk 2 above)
Comment (d2r) [inline storm-core/src/clj/org/apache/storm/testing.clj]: This is the only place it remains, so I'll leave it here. When testing.clj is translated, we can leave the macros until all namespaces using it are removed.

File: storm-core/src/clj/org/apache/storm/stats.clj (hunk 1 above)
Comment (d2r) [inline storm-core/src/clj/org/apache/storm/stats.clj]: Actually, there are only two spots where this is needed, so leaving it in.

**Comment by rfarivar** [comment]
+1

**Pull request merged**